### ISSUE TITLE
User Account – Page crash on `subscription.incident_id.incident_id`

### ISSUE THREAD

**Issue by kepae** [responsible-ai-collaborative/aiid#2364]
When @datherton09 is signed in and clicks on the "Your Account" page, it begins to render and then fails to a completely blank page. I don't have dev console logs but from Rollbar, I found a corresponding error in Rollbar:

https://app.rollbar.com/a/seanbmcgregor/fix/item/AIID/5198

`webpack://gatsby-gitbook-aiid/./src/components/UserSubscriptions.js:125`
`incidentId: subscription.incident_id.incident_id,`
`TypeError Cannot read properties of null (reading 'incident_id')`

This error comes from this line:
https://github.com/responsible-ai-collaborative/aiid/blob/0c0ec744c4216e79f019e99cae936749cea1e773/site/gatsby-site/src/components/UserSubscriptions.js#L125

I believe the subscription object is not being destructured correctly. It _may_ be that the logic doesn't handle different subscription types where `incident_id` doesn't exist:
https://github.com/responsible-ai-collaborative/aiid/blob/0c0ec744c4216e79f019e99cae936749cea1e773/site/gatsby-site/src/utils/subscriptions.js#L6
Danny's user account is attached to a subscription of type `new-incident` that doesn't have an `incident_id` field, but so is mine, so I'm not sure where the error lies.

Actions:
* Fix broken subscription listing behavior.
* Review and add test for this.
* This bug should not have resulted in the page completely failing. We should review user account page for other errors that cause page crashes.

**Comment by pdcp1** [comment]
@kepae Do you know the steps to reproduce the issue?
I followed the classic sign-in flow on Staging and Productions and I couldn't reproduce it.

1. Signup with an email/password on `/signup` page
2. I received the confirmation email and I confirmed the account
3. Login on `/login`
4. Go to the Account `/account` page
5. The account page loads successfully without any error

**Comment by kepae** [comment]
I can't reproduce the issue myself -- it seems specific to @datherton09's account.

Possibly, could you try to reproduce by copying the subscriptions into staging to a new user?

**Comment by pdcp1** [comment]
I found that there is a subscription to this Incident https://incidentdatabase.ai/cite/542 which was marked (manually) as duplicated, so when we try to get the Incident information, it fails.
I added a check to the subscription section to avoid this situation.
The actual fix for this is to delete all subscription to an specific Incident as soon the Incident is marked as duplicated. It should be handle as part of this PR https://github.com/responsible-ai-collaborative/aiid/pull/2130
